### DISCUSSION TITLE
crt-scanlines.effect MacOS

### DISCUSSION THREAD

**Discussion by Fafnyir** [Xaymar/obs-stream-effects#976] [General]
Greetings all, and my apologies if this has already been answered.

I've been trying to get the "ScanLines" fader to work on MacOS but am having no luck. Same version installed on windows works fine. Is there a MacOS bug or is it a graphics card issue maybe?

**Comment by Xaymar** [comment]
This is due to libOBS's shader transpiler not correctly porting all language features from OBS-SL (a subset of HLSL) to GLSL. Your log file usually contains what the error is, and will explain if it's a missing GPU feature, or a shader needing special patches.

Though as you can probably guess, the provided shaders are examples not meant for direct usage.

**Comment by Fafnyir** [comment]
Log file shows ```
10:24:34.282: device_pixelshader_create (GL) failed
10:24:34.282: Pass (0) <> missing pixel shader!
10:24:34.283: [StreamFX] Loading shader '/Users/user/Library/Application Support/obs-studio/plugins/StreamFX.plugin/Contents/Resources/examples/shaders/filter/crt-scanlines.effect' failed with error: Unknown error during effect compile.
```

**Reply by Xaymar** [reply]
The log file has more information than just that. Just upload the file instead of a small section of it.
